### PR TITLE
Migrate InstrumentTypes to DX

### PR DESCRIPTION
**Pull request by Bugerman58**
## Description of the issue/feature this PR addresses

This PR migrates the AT InstrumentType(s) to Dexterity types for Python 3 compatibility.

## Current behavior before PR

InstrumentType(s) are content types based on the Archetypes framework

## Desired behavior after PR is merged

InstrumentType(s) are content types based on the Dexterity framework

--
I confirm I have tested this PR thoroughly and coded it according to [PEP8][1]
and [Plone's Python styleguide][2] standards.

[1]: https://www.python.org/dev/peps/pep-0008
[2]: https://docs.plone.org/develop/styleguide/python.html

### CODE REVIEW

**Comment by Bugerman58** [comment]
Hi @xispa and @ramonski 
this PR ready for review.

File: src/senaite/core/browser/controlpanel/instrumenttypes/view.py [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+        self.title = translate(_(
+            "listing_instrumenttypes_title",
+            default="Instrument Types")
+        )
+        self.description = ""
+        self.icon = "{}/{}".format(
+            self.portal_url,
+            "/++resource++bika.lims.images/instrumenttype_big.png"
+        )
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/instrumenttypes/view.py]: Use the default icon provided by the type:

```python
self.icon = api.get_icon("InstrumentTypes", html_tag=False)
```

File: src/senaite/core/browser/controlpanel/instrumenttypes/view.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright 2018-2024 by it's authors.
+# Some rights reserved, see README and LICENSE.
+
+import collections
+
+from bika.lims import api
+from bika.lims import bikaMessageFactory as _
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/instrumenttypes/view.py]: Better use `senaiteMessageFactory`:

```python
from bika.lims import senaiteMessageFactory as _
```

File: src/senaite/core/browser/controlpanel/instrumenttypes/view.py [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+            "listing_instrumenttypes_title",
+            default="Instrument Types")
+        )
+        self.description = ""
+        self.icon = "{}/{}".format(
+            self.portal_url,
+            "/++resource++bika.lims.images/instrumenttype_big.png"
+        )
+
+        self.show_select_row = False
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/instrumenttypes/view.py]: Remove `self.show_select_row`, this attribute was removed from the base class

File: src/senaite/core/browser/controlpanel/instrumenttypes/view.py [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        self.show_select_column = True
+        self.pagesize = 25
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/instrumenttypes/view.py]: Remove `self.pagesize` to stick to [default's (50)](https://github.com/senaite/senaite.app.listing/blob/2.x/src/senaite/app/listing/view.py#L173)

File: src/senaite/core/browser/controlpanel/instrumenttypes/view.py [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+                "permission": AddInstrumentType,
+                "icon": "senaite_theme/icon/plus"
+            }
+        }
+
+        self.title = translate(_(
+            "listing_instrumenttypes_title",
+            default="Instrument Types")
+        )
+        self.description = ""
```
Comment (xispa) [inline src/senaite/core/browser/controlpanel/instrumenttypes/view.py]: Remove `self.description = ""`, it is already set to empty in the base class

File: src/senaite/core/upgrade/v02_06_000.zcml [hunk 6]
```diff
@@ -3,23 +3,31 @@
     xmlns:genericsetup="http://namespaces.zope.org/genericsetup"
     i18n_domain="senaite.core">
 
-    <genericsetup:upgradeStep
+  <genericsetup:upgradeStep
+      title="SENAITE.CORE 2.6.0: Migrate Instrument Types to DX"
+      description="Migrate Instrument Types to Dexterity"
+      source="2629"
+      destination="2630"
```
Comment (xispa) [inline src/senaite/core/upgrade/v02_06_000.zcml:10]: Update `profiles/metadata.xml` in accordance

**Review by xispa — CHANGES_REQUESTED** [review]
Thanks @Bugerman58 !, some minor changes required!

**Review by xispa — APPROVED** [review]

**Pull request merged**